shell/platform/windows/flutter_windows_view_unittests.cc

```diff
@@ -209,12 +209,216 @@ TEST(FlutterWindowsEngine, AddSemanticsNodeUpdate) {
+//
+//        node0
+//        /   \
+//    node1    node2
+//               |
+//             node3
+//
+// node0 and node2 are grouping nodes. node1 and node2 are static text nodes.
+TEST(FlutterWindowsEngine, AddSemanticsNodeUpdateWithChildren) {
+  std::unique_ptr<FlutterWindowsEngine> engine = GetTestEngine();
+  EngineModifier modifier(engine.get());
+  modifier.embedder_api().UpdateSemanticsEnabled =
+      [](FLUTTER_API_SYMBOL(FlutterEngine) engine, bool enabled) {
+        return kSuccess;
+      };
+
+  auto window_binding_handler =
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>();
+  FlutterWindowsView view(std::move(window_binding_handler));
+  view.SetEngine(std::move(engine));
+
+  // Enable semantics to instantiate accessibility bridge.
+  view.OnUpdateSemanticsEnabled(true);
+
+  auto bridge = view.GetEngine()->accessibility_bridge().lock();
+  ASSERT_TRUE(bridge);
+
+  // Add root node.
+  FlutterSemanticsNode node0{sizeof(FlutterSemanticsNode), 0};
+  std::vector<int32_t> node0_children{1, 2};
+  node0.child_count = node0_children.size();
+  node0.children_in_traversal_order = node0_children.data();
+  node0.children_in_hit_test_order = node0_children.data();
+
+  FlutterSemanticsNode node1{sizeof(FlutterSemanticsNode), 1};
+  node1.label = "prefecture";
+  node1.value = "Kyoto";
+  FlutterSemanticsNode node2{sizeof(FlutterSemanticsNode), 2};
+  std::vector<int32_t> node2_children{3};
+  node2.child_count = node2_children.size();
+  node2.children_in_traversal_order = node2_children.data();
+  node2.children_in_hit_test_order = node2_children.data();
+  FlutterSemanticsNode node3{sizeof(FlutterSemanticsNode), 3};
+  node3.label = "city";
+  node3.value = "Uji";
+
+  bridge->AddFlutterSemanticsNodeUpdate(&node0);
+  bridge->AddFlutterSemanticsNodeUpdate(&node1);
+  bridge->AddFlutterSemanticsNodeUpdate(&node2);
+  bridge->AddFlutterSemanticsNodeUpdate(&node3);
+  bridge->CommitUpdates();
+
+  // Look up the root windows node delegate.
+  auto node_delegate = bridge
+                           ->GetFlutterPlatformNodeDelegateFromID(
+                               AccessibilityBridge::kRootNodeId)
+                           .lock();
+  ASSERT_TRUE(node_delegate);
+  EXPECT_EQ(node_delegate->GetChildCount(), 2);
+
+  // Get the native IAccessible object.
+  IAccessible* node0_accessible = node_delegate->GetNativeViewAccessible();
+  ASSERT_TRUE(node0_accessible != nullptr);
+
+  // Property lookups will be made against this node itself.
+  VARIANT varchild{};
+  varchild.vt = VT_I4;
+  varchild.lVal = CHILDID_SELF;
+
+  // Verify node type is a group.
+  VARIANT varrole{};
+  varrole.vt = VT_I4;
+  ASSERT_EQ(node0_accessible->get_accRole(varchild, &varrole), S_OK);
+  EXPECT_EQ(varrole.lVal, ROLE_SYSTEM_GROUPING);
+
+  // Verify child count.
+  long node0_child_count = 0;
+  ASSERT_EQ(node0_accessible->get_accChildCount(&node0_child_count), S_OK);
+  EXPECT_EQ(node0_child_count, 2);
+
+  {
```

Comment:
why additional scope?

Reply:
Just to keep from shadowing variables (the variables for reading name/value/role).

Reply:
this looks good :), was just wondering whether this is part of the test.